pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -272,11 +275,28 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 						CacheDuration: &durationpb.Duration{Seconds: 5 * 60},
 					},
 				}
-			} else {
+			} else if features.JwksFetchMode == jwt.Hybrid {
 				provider.JwksSourceSpecifier = push.JwtKeyResolver.BuildLocalJwks(jwtRule.JwksUri, jwtRule.Issuer, "")
+			} else {
+				// Log error and create remote JWKs with fake cluster
+				authnLog.Errorf("Failed to look up Envoy cluster %v. "+
+					"Please create ServiceEntry to register external JWKs server or "+
+					"set PILOT_JWT_ENABLE_REMOTE_JWKS to hybrid/istiod mode.", err)
+				provider.JwksSourceSpecifier = &envoy_jwt.JwtProvider_RemoteJwks{
+					RemoteJwks: &envoy_jwt.RemoteJwks{
+						HttpUri: &core.HttpUri{
+							Uri: jwtRule.JwksUri,
+							HttpUpstreamType: &core.HttpUri_Cluster{
+								Cluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", jwksInfo.Hostname, jwksInfo.Port),
```

Comment:
`jwksInfo.Hostname` is not quite right, it could be `namespace/hostname` format. See LookupCluster

Reply:
I think multiple forms are accepted, see https://github.com/istio/istio/blob/8ce0defcc905873dadf3fa1d8c3f3629cd39895c/pilot/pkg/extensionproviders/cluster_lookup.go#L32 which is parsing it?

=

Reply:
Can jwksUri be configured as `ns/service`?

Reply:
idk, i assumed so since the function handles it... maybe that is only for other use cases though

Reply:
Don't think jwksUri can be configured as `ns/svc`, if it is in-cluster, it will be `foo.ns.svc.cluster.local`.